src/nnet3/nnet-utils.cc

```diff
@@ -40,6 +40,106 @@ int32 NumInputNodes(const Nnet &nnet) {
   return ans;
 }
 
+int32 GetTModulusForIvector(const Nnet &nnet) {
+  std::vector<std::string> name;
+  name.push_back("ivector");
+  std::vector<int32> value;
```

Comment:
this variable should have a plural name. It is difficult to understand what values you are getting. It is better to choose a descriptive variable name, say <code>periods</code> ?